### PR TITLE
Feature: Strict mode

### PR DESCRIPTION
**Pull request by serradura**

### CODE REVIEW

File: lib/ruby/lambdas/core/strict_mode/disable.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[:String!, :Numeric!].each do |name|
+  Kernel.remove_method(name) if Kernel.respond_to?(name)
+end
```
Comment (imteekay) [inline lib/ruby/lambdas/core/strict_mode/disable.rb:9]: I think it's good to add a description or something like to show that this methods will be overwritten by new ones. I don't know if we add this to the readme, or some king of documentation

File: examples/strict_mode/01.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# ---
+
+require "ruby/lambdas/strict-mode"
+
+p [2, 4, 8].map(&Numerics + 2) # => [4, 6, 10]
+
+begin
+  p ["2", "4", "8"].map(&Numerics + "2")
+rescue TypeError => e
+ p e # => #<TypeError: "2" must be Numeric>
```
Comment (imteekay) [inline examples/strict_mode/01.rb]: Missing space

```suggestion
  p e # => #<TypeError: "2" must be Numeric>
```

File: examples/strict_mode/01.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+end
+
+p ["2 ", " 4", "8 "].map(&Strings::Strip) # => ["2", "4", "8"]
+
+begin
+  p [2, 4, 8].map(&Strings::Strip)
+rescue TypeError => e
+ p e # => #<TypeError: 2 must be a String>
```
Comment (imteekay) [inline examples/strict_mode/01.rb]: Missing space

```suggestion
  p e # => #<TypeError: 2 must be a String>
```

File: lib/ruby/lambdas/core/strict_mode/disable.rb (hunk 1 above)
Comment (serradura) [reply lib/ruby/lambdas/core/strict_mode/disable.rb:9]: @leandrotk what do you think about this short explanation?
```ruby
# These Kernel methods can be overridden by 
# new ones if the strict mode was enabled. e.g:
# require "ruby/lambdas/strict-mode"
```

Comment (imteekay) [reply lib/ruby/lambdas/core/strict_mode/disable.rb:9]: It's great!

**Review by imteekay — APPROVED** [review]

**Pull request merged**